app/controllers/playlist_controller.rb

```diff
@@ -9,9 +9,18 @@ def show
     @playlist = Playlist.where(hashed_id=session[hashed_id])
     @hashed_id = session[:hashed_id]
     if params[:search]
-      @songs = Song.where("lower(name) LIKE ? OR lower(artist) LIKE ?", "%#{params[:search].downcase}%", "%#{params[:search].downcase}%").where( id: PlaylistSong.where(hashed_id: @hashed_id).pluck(:song_id) ).order('vote_count DESC')
+      # deprecated
+      # @songs = Song.where("lower(name) LIKE ? OR lower(artist) LIKE ?", "%#{params[:search].downcase}%", "%#{params[:search].downcase}%").where( id: PlaylistSong.where(hashed_id: @hashed_id).pluck(:song_id) ).order('vote_count DESC')
+
+      # updated, valid
+      @songs = Song.where("lower(name) LIKE ? OR lower(artist) LIKE ?", "%#{params[:search].downcase}%", "%#{params[:search].downcase}%").joins(:playlist_songs).where(playlist_songs:{hashed_id: session[:hashed_id] }).select("songs.*, playlist_songs.vote_count").order('vote_count DESC')
```

Comment:
One thing I noticed while implementing the play songs feature is we can't play albums or artists names. So i changed our search query to return just tracks because those are actually playable. My reasoning behind this is: why have a song on a queue if it isn't a song.

Reply:
This may affect line 16